samtranslator/validator/validator.py

```diff
@@ -7,49 +9,236 @@
 
 
 class SamTemplateValidator(object):
-    @staticmethod
-    def validate(template_dict, schema=None):
+    """
+    SAM template validator
+    """
+
+    # Useful to find a unicode prefixed string type to replace it with the non unicode version
+    # On Python2, the validator returns types in error messages prefixed with 'u'
+    # We remove them to be consistent between python versions
+    # Example: "u'integer'" -> "'integer'"
+    UNICODE_TYPE_REGEX = re.compile("u('[^']+')")
+
+    def __init__(self, schema_path=None):
+        """
+        Constructor
+
+        Parameters
+        ----------
+        schema_path : str, optional
+            Path to a schema to use for validation, by default None, the default schema.json will be used
+        """
+        if not schema_path:
+            schema_content = self._read_default_schema()
+        else:
+            schema_content = self._read_json(schema_path)
```

Comment:
Suggestion:

```
if schema_path is None:
   schema_path = "default path"

schema_content = self._read_json(schema_path)
```

Not sure if that works, but I like minimizing control statements where possible.

Reply:
Yes!
Don't remember why I went with this form, I was maybe reusing `_read_default_schema` elsewhere before.